app/Traits/AccessControlAPI.php

```diff
@@ -58,55 +63,48 @@ public function accessControl($api_key, $control_table = 'filesets')
-            // content associated with the API key. For this case, the query has been optimized,
-            // and returns a list of language ids instead of a list of fileset hashes
-            $dbp_users = config('database.connections.dbp_users.database');
-            $dbp_prod = config('database.connections.dbp.database');
-            switch ($control_table) {
-                case 'languages':
-                    $identifiers = DB::select(
-                        DB::raw(
-                            'select distinct l.id identifier
-                            from ' . $dbp_users . '.user_keys uk
-                            join ' . $dbp_users . '.access_group_api_keys agak on agak.key_id = uk.id
-                            join ' . $dbp_prod . '.access_group_filesets agf on agf.access_group_id = agak.access_group_id
-                            join ' . $dbp_prod . '.bible_fileset_connections bfc on agf.hash_id = bfc.hash_id
-                            join ' . $dbp_prod . '.bibles b on bfc.bible_id = b.id
-                            join ' . $dbp_prod . '.languages l on l.id = b.language_id
-                            where uk.id = ?'
-                        ),
-                        [$key->id]
-                    );
-                    break;
-                case 'bibles':
-                    $identifiers = DB::select(
-                        DB::raw(
-                            'select distinct bfc.bible_id identifier
-                            from ' . $dbp_users . '.user_keys uk
-                            join ' . $dbp_users . '.access_group_api_keys agak on agak.key_id = uk.id
-                            join ' . $dbp_prod . '.access_group_filesets agf on agf.access_group_id = agak.access_group_id
-                            join ' . $dbp_prod . '.bible_fileset_connections bfc on agf.hash_id = bfc.hash_id
-                            where uk.id = ?'
-                        ),
-                        [$key->id]
-                    );
-                    break;
-                default:
-                    // Use Eloquent everywhere except for this giant request
-                    $identifiers = AccessGroupFileset::select('hash_id as identifier')
-                        ->whereIn('access_group_id', $accessGroups->pluck('id'))->distinct()->get();
-                    break;
-            }
+                // Access Control has historically been tied to fileset hashes.
+                // As the number of filesets grows, this has been affected query
+                // performance for endpoints such as Languages (and similarly for  Bibles),
+                // which return a list of Languages associated with any fileset
+                // content associated with the API key. For this case, the query has been optimized,
+                // and returns a list of language ids instead of a list of fileset hashes
+                // $dbp_users = config('database.connections.dbp_users.database');
```

Comment:
remove dbp_users and dbp_prod commented lines

Reply:
@bradflood , it has been removed.

Also, I have done a refactor about the `AccessControlAPI` class to make sure that it will use  the AccessGroups ids that it fetches from AccessControl middleware rather that it uses the `api_user_key` to validate the fileset permissions. You can see the changes in the following commit.

https://github.com/faithcomesbyhearing/dbp/pull/904/commits/c6f5d6a7e6f693ab564b4255965c35ab74776095#diff-0e4323e6a03c91c9c8dc88afef15fdad09636468c0ac3550f497618372788b6d